util/winutil/winutil_windows.go

```diff
@@ -225,9 +225,20 @@ func isSIDValidPrincipal(uid string) bool {
 	}
 }
 
-// EnableCurrentThreadPrivilege enables the named privilege
-// in the current thread access token.
+// EnableCurrentThreadPrivilege enables the named privilege in the current
+// thread access token. Callers should use DisableCurrentThreadPrivilege when
+// done with the privileged task.
 func EnableCurrentThreadPrivilege(name string) error {
+	return adjustCurrentThreadPrivilege(name, windows.SE_PRIVILEGE_ENABLED)
+}
+
+// DisableCurrentThreadPrivilege disabled the named privilege in the current
+// thread access token.
+func DisableCurrentThreadPrivilege(name string) error {
+	return adjustCurrentThreadPrivilege(name, windows.SE_PRIVILEGE_REMOVED)
+}
+
+func adjustCurrentThreadPrivilege(name string, attr uint32) error {
```

Comment:
It doesn't really matter for our current sole user, but if this ever becomes something we do more hotly, having Enable return a `disable func()` would let us do the UTF16PtrFromString and LookupPrivilegeValue just once.

Also, then the Enable can do its own runtime.LockOSThread() (they nest nowadays!) that the disable/release func could undo, removing that potential footgun for callers who havent' already done the LockOSThread.

If not, at least document callers must LockOSThread on their own? (it looks like?)

Reply:
Moved ImpersonateSelf/LockOSThread into the function.
Per @dblohm7's comment, we can use `RevertToSelf` to drop the privilege, no need to explicitly disable it.